extension/ackextension/go.mod

```diff
@@ -8,6 +8,7 @@ require (
 	go.opentelemetry.io/collector/extension v0.96.1-0.20240306115632-b2693620eff6
 	go.opentelemetry.io/otel/metric v1.24.0
 	go.opentelemetry.io/otel/trace v1.24.0
+	go.uber.org/goleak v1.3.0
```

Comment:
Why do we need this? we aren't spawning any goroutines right? Are you thinking of testing with `-race` flag to test race conditions?

Reply:
This is added as a part of my request, that `goleak` be added to help ensure no goroutines are being leaked. We're working on requiring this to be a test for every component in the collector, it's nothing specific to this component 👍

Reply:
ah ok got it.